[16.0][MIG] account_check_deposit

alexis-via opened the pull request.

bguillot committed:
[INIT] init module to manage check deposits

sebastienbeau committed:
[FIX] fix remove pdb call

chafique-delli committed:
[IMP]account_check_deposit: add the print option webkit

bguillot committed:
[FIX+IMP] account_check_deposit: Several things:

* Clean code
* Add access rules for account_check_deposit
* Change print button name

sebastienbeau committed:
[FIX+REF+IMP] account_check_deposit:

* [FIX] fix datamodel structure, indeed check_payment_ids should be a m2o. In the view we still keep the widget
* [REF] remove useless state cancel, only draft/done is sufficent for now, forbid to delete a done deposit, fix move_line name need to be the equal to the line.ref for the reconciliation
* [IMP] improve reconciliation between the check and the check deposit
* [FIX] set correctly the date on the move line generated
* [FIX] fix reconciliation of check
* [IMP] check deposit improve view and add tab for move line generated
* [IMP] add total amount and flag to know if the deposit have been reconcile, the best will be to add a workflow
* [FIX] fix error when replaying commit
* [REF] remove dependency on sale_quick_payment as check can be registred in various way like native voucher
* [CLEAN] pep8 cleanup no code change
* [REF] clean remove useless code, default value are called during the create
* [FIX] fix wrong spelling
* [REF] V7 clean up

Commit:
Add support for multi-currency
Works with a single account journal
Usability and many other enhancements

alexis-via committed:
Port to v8
Convert webkit report to qweb

sebastienbeau committed:
[FIX] fix domain, give the posibility to use a different journal (one for the payment, one for the deposit), hide unreconcile entry which a link to a deposit

rschnapka committed:
german translations upto invoice currency

legalsylvain committed:
[FIX] move translation file in unported folders

sebastienbeau committed:
[PORT] start porting account_check_deposit

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[REF] account_check_deposit : Port to new API

* Move description from __openerp__.py to README.rst
* Move files in sub-directories
* FIX access rights issue when user was not accounting manager

OCA Transbot updated translations from Transifex

mourad-ehm committed:
[MIG] account_check_deposit: Migration to 9.0

* create move and move line at the same time
* add tests

florian-dacosta committed:
Fix payment field + automatic reconciliation on validation

alexis-via committed:
[MIG] account_check_deposit: Migration to v10

* Add option to have the counter-part of the deposit move directly to the bank account
  Use account.config.settings page
* Remove data that create a bank journal because it blocks the loading of the chart of accounts when
  the module is installed at the same time as the chart of account (via dependencies)

OCA Transbot updated translations from Transifex

alexis-via committed:
[FIX+imp] account_check_deposit: no auto-post move + use the bank journal if the offsetting account is the bank account

OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] account_check_deposit: Change company currency in tests by SQL

For avoiding constraint.

[UPD] Update account_check_deposit.pot

pedrobaeza committed:
[MIG] account_check_deposit: Migration to 11.0

[UPD] Update account_check_deposit.pot

[UPD] README.rst

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (61 of 61 strings)

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_check_deposit/de/

Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

ThomasBinsfeld committed:
[MIG] Account Check Deposit to 12.0

alexis-via committed:
account_check_deposit: Don't force 'name' of account.move

Add print button (feature up-ported from v10)
Add option to post moves (feature up-ported from v10)
Sequence now takes into account the deposit date
Customize the filename of the report
Update README with v12 labels

[UPD] README.rst

[UPD] Update account_check_deposit.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/

[UPD] README.rst

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (63 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/pt/

HviorForgeFlow committed:
[12.0] account_financial_tools: Lint fixing

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (63 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (63 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/pt/

ps-tubtim committed:
[IMP] account_check_deposit: black, isort

ps-tubtim committed:
[MIG] account_check_deposit: Migration to 13.0

oca-travis committed:
[UPD] Update account_check_deposit.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_check_deposit/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_check_deposit/

alexis-via committed:
[MIG] account_check_deposit from v13 to v14

alexis-via committed:
Remove post_move option, because we cannot reconcile non-posted entries, so we should always post the move

alexis-via committed:
account_check_deposit: remove configuration parameters from res.company

Update documentation
Don't duplicate labels between name of account.move.line and ref of
account.move, now that name of account.move.line is not required any
more.

alexis-via committed:
account_check_deposit: usability improvements

Add button "Get All Received Checks"
Add default value for journal_id and bank_journal_id when there is only 1 possible value
Add hide/show options on lines
Improve multi-company handling

alexis-via committed:
account_check_deposit: migrate tests

alexis-via committed:
account_check_deposit: Add chatter on check deposit

Add chatter message when you click on the button to get all checks
Add unicity SQL constraint

[UPD] Update account_check_deposit.pot

[UPD] README.rst

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 84.3% (54 of 64 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/fr/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (64 of 64 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/pt/

[UPD] Update account_check_deposit.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (90 of 90 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/pt/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 67.7% (61 of 90 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/fr/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 70.0% (63 of 90 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/it/

MiquelRForgeFlow committed:
[FIX] Avoid warning

account_check_deposit 14.0.1.0.1

[IMP] update dotfiles

alexis-via committed:
[MIG] account_check_deposit from v14 to v15

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 72.2% (65 of 90 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 74.4% (67 of 90 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/it/

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 17.7% (16 of 90 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_check_deposit/pt_BR/

alexis-via committed:
check_deposit: filter source moves on state=posted

account_check_deposit 14.0.1.1.0

victoralmau committed:
[MIG] account_check_deposit: Migration to 15.0

TT36937

Commit:
[UPD] Update account_check_deposit.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-15.0/account-financial-tools-15.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-15-0/account-financial-tools-15-0-account_check_deposit/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (89 of 89 strings)

Translation: account-financial-tools-15.0/account-financial-tools-15.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-15-0/account-financial-tools-15-0-account_check_deposit/es_AR/

alexis-via committed:
[MIG] account_check_deposit to v16

alexis-via committed:
account_check_deposit: re-write move generation

In v16, if we create the account.move and then the account.move.line
one-by-one, we end up with an additionnal line with credit=debit=0 and
label="Automatic balancing line". So I decided to re-write move
generation and simplify the move of the check deposit: no need to have
one move line per check, we can have a single line for the total of checks
without partner and reconcile this single line with all the checks.

legalsylvain commented:
Coule you take a look on pending old PR and cherry pick various improvments ?
Thanks 
https://github.com/OCA/account-financial-tools/pull/976
https://github.com/OCA/account-financial-tools/pull/872

legalsylvain commented:
Cc : @ivantodorovich, @mourad-ehm

ivantodorovich commented on account_check_deposit/__manifest__.py:
```diff
@@ -1,13 +1,13 @@
-# Copyright 2012-2021 Akretion France (http://www.akretion.com/)
+# Copyright 2012-2022 Akretion France (http://www.akretion.com/)
```
FTR, there's no need to specify an "end date" for copyright.
You could just write it like this and forget of updating it ever again:

`# Copyright 2012 Akretion France (http://www.akretion.com/)`

sergiocorato replied:
> FTR, there's no need to specify an "end date" for copyright. You could just write it like this and forget of updating it ever again:
> 
> `# Copyright 2012 Akretion France (http://www.akretion.com/)`

This review comment is pending too.

ivantodorovich reviewed (approved):
LGTM

legalsylvain reviewed (changes requested):
Need answers regarding pending unreviewed PRs.

alexis-via committed:
account_check_deposit: replace api.onchange by computed fields

Add an error message in case of configuration error
Use official translate method for multi-string replacements in error messages

alexis-via commented:
I have a suggestion: what about allowing users from the group "account.group_account_invoice" to use the check deposit feature, and not just users that belong to "account.group_account_user". Making checks deposits is a pretty basic task that is often handled by an accounting secretary and not by the accountant. It would imply moving the menu entry from "Invoicing > Accounting" to "Invoicing > Customers".
If nobody is against this idea, I will commit the change in this PR.

alexis-via committed:
check_deposit: allow account.group_account_invoice to make checks deposits

Add confirm pop-up when clicking on "Back to draft"

alexis-via commented:
As proposed in my previous comment, members of account.group_account_invoice" can now make a check deposit, and I moved the menu entry to "Invoicing > Customers". If someone is against the idea and didn't have the time to comment, you can still give your argument and I may revert it.
For better menu entry ordering in "Invoicing > Customers", I made this other PR https://github.com/OCA/credit-control/pull/320/files

alexis-via committed:
account_check_deposit: update FR translation

Fix a few strings

alexis-via committed:
[IMP] account_check_deposit: add ACL for account read-only group

gregory-moka commented:
@alexis-via 
This PR seems ok, what should be done ?

PlantBasedStudio commented:
This PR has been pending for several months, no one to merge it?

sergiocorato commented:
> Coule you take a look on pending old PR and cherry pick various improvments ? Thanks #976 #872

This comment is pending.

njeudy commented:
@alexis-via can we merge this ...

alexis-via commented:
@njeudy I would love to! @legalsylvain made a review with required changes, so it blocks the migration. The changes in PR #976 are already in the code base of the module in v16. But not the changes introduced by PR #872 ; I don't know if they are still relevant.

legalsylvain commented:
> But not the changes introduced by PR https://github.com/OCA/account-financial-tools/pull/872 ; I don't know if they are still relevant.

Yes. It looks still relevant. 

Thanks !

legalsylvain commented:
Hi @alexis-via. No news from this PR.

Please ping me if you have trouble to cherry-pick #872.

Thanks !
